saskatoon/harvest/api.py

```diff
@@ -221,3 +224,188 @@ def list(self, request, *args, **kwargs):
                                  'title': _("New Person")
                                  }
                          })
+
+
+class StatsViewset(LoginRequiredMixin, viewsets.ModelViewSet):
+
+    template_name = "app/stats.html"
+    queryset = Harvest.objects.filter(status="Succeeded").order_by("-id")
+    permission_classes = [IsCoreOrAdmin]
+    serializer_class = HarvestSerializer
+    filter_backends = (filters.DjangoFilterBackend,)
+    filterset_class = HarvestFilter
+    filterset_fields = (
+        "pick_leader",
+        "owner_fruit",
+        "property__neighborhood",
+        "property",
+        "status",
+        "season",
+    )
+    # ? season name that will be passed to template if no season is specified
+    season_name = "all"
+    # ? seasons that will be shown in the dropdown list
+    # TODO Use the original seasons ChoiceField in dropdown
+    seasons = list(map(lambda x: x[0], filterset_class.seasons))
+
+    def list(self, request, format="html", *args, **kwargs):
+        """returns statistics on harvests for all seasons or a specific season"""
+
+        season_parameter = self.kwargs.get("season")
+        if season_parameter is not None:
+            if season_parameter in self.seasons:
+                self.queryset.filter(start_date__year=season_parameter)
+            else:
+                raise Http404("Season Not found")
+
+        response = super(StatsViewset, self).list(request, *args, **kwargs)
+        if request.accepted_renderer.format == "json":
+            return Response(response.data)
+
+        highlights = self.get_highlights()
+        total_fruit = self.get_total_weight_per_fruit()
+        total_neighborhood = self.get_total_weight_per_neighborhood()
+        total_beneficiary = self.get_total_weight_per_beneficiary()
+        total_picker = self.get_total_weight_per_picker()
+        season_name = self.season_name if season_parameter is None else season_parameter
+
+        return Response(
+            {
+                "filter": get_filter_context(self),
+                "harvests": response.data,
+                "season": season_name,
+                "seasons": self.seasons,
+                "highlights": highlights,
+                "total_fruit": total_fruit,
+                "total_neighborhood": total_neighborhood,
+                "total_beneficiary": total_beneficiary,
+                "total_picker": total_picker,
+            }
+        )
+
+    def get_highlights(self):
+        total_weight = int(
+            HarvestYield.objects.aggregate(Sum("total_in_lb")).get("total_in_lb__sum")
+        )
+        total_harvests = Harvest.objects.count()
+        total_pickers = HarvestYield.objects.values("recipient").distinct().count()
+        total_beneficiaries = Organization.objects.filter(is_beneficiary=True).count()
+
```

Comment:
@mina-andajos I'm a bit confused by the effect of the `season` filter. When we pass a season as kwarg, we should expect stats to be filtered over that season right?